provider/datagen/src/baked_exporter.rs

```diff
@@ -466,8 +492,11 @@ impl BakedExporter {
 
         let maybe_msrv = maybe_msrv();
 
-        let body = if values.is_empty() {
-            quote!(Err(icu_provider::DataErrorKind::MissingLocale.with_req(<#marker as icu_provider::KeyedDataMarker>::KEY, req)))
+        let (load_body, iterable_body) = if values.is_empty() {
+            (
+                quote!(Err(icu_provider::DataErrorKind::MissingLocale.with_req(<#marker as icu_provider::KeyedDataMarker>::KEY, req))),
+                quote!(Err(icu_provider::DataErrorKind::MissingLocale.with_req(<#marker as icu_provider::KeyedDataMarker>::KEY, req))),
```

Comment:
`iterable_body` should be `Ok(vec![])` here

Reply:
In order to use `vec!` I'd need to repeat the `<icu_provider::DataLocale as core::str::FromStr>::from_str(s).unwrap()` for each element of the vector. I don't think there is a performance penalty with the way the code is written now because [`[T; N].into_iter()` implements `ExactSizeIterator`](https://doc.rust-lang.org/src/core/array/iter.rs.html#365) so it should be able to efficiently create the vector.

Reply:
What I'm saying is that it should not return `Err`, it should return `Ok(vec![])`. `supported_locales` should *never* return `MissingLocale`, because we don't ask it for a locale.

Reply:
Ok I see. Fixed.